authors/apps/profiles/models.py

```diff
@@ -0,0 +1,48 @@
+from django.db import models
+from django.dispatch import receiver
+from django.db.models.signals import post_save
+from authors.settings import AUTH_USER_MODEL as User
+from cloudinary.models import CloudinaryField
+
+
+class Profile(models.Model):
+    """
+    This model creates a user profile with bio and
+    image field once a user creates an account.
+    """
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
+    bio = models.TextField(default="Update your bio description")
+    image = CloudinaryField('image')
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    @property
+    def image_url(self):
+        """
+        Get the image url. This spits a url that resolves to the image.
+        """
+        return self.image.url
+
+    @property
+    def username(self):
+        return self.user.username
+
+    def __str__(self):
+        return self.user.username
+
+
+@receiver(post_save, sender=User)
+def create_user_profile(sender, instance, created, **kwargs):
+    """
+    This is triggered when a user is created in order to create their profile.
+    """
+    if created:
+        Profile.objects.create(user=instance)
```

Comment:
This statement will create and save the profile right??

Reply:
Yes, I had repeated myself.

Reply:
From my understanding, it just creates a profile instance that is yet to be committed to the database. `save_user_profile` is what will actually save the profile. At this point, we are sure that it can now be saved since the user model has been persisted. I think `save_user_profile` method alone is enough to create and save the profile @charisschomba.

Reply:
@mutaimwiti I tested without the second block and it works too